File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/watchproviders/WatchProviderPaths.java
package dev.reuss.tmdb.domain.watchproviders;

final class WatchProviderPaths {

    private static final String WATCH_PROVIDER = "/watch/providers";

    private WatchProviderPaths() {
    }

    static String availableRegions() {
        return WATCH_PROVIDER + "/regions";
    }

    static String movieProviders() {
        return WATCH_PROVIDER + "/movie";
    }

    static String tvProviders() {
        return WATCH_PROVIDER + "/tv";
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/watchproviders/WatchProviderPaths.kt
package dev.reuss.tmdb.domain.watchproviders

import dev.reuss.tmdb.core.path.tmdbPath

internal object WatchProviderPaths {

    fun availableRegions(): String =
        tmdbPath("watch", "providers", "regions")

    fun movieProviders(): String =
        tmdbPath("watch", "providers", "movie")

    fun tvProviders(): String =
        tmdbPath("watch", "providers", "tv")
}